--- vsock-only/linux/af_vsock.c	2015-08-04 18:26:51.705735963 -0700
+++ vsock-only/linux/af_vsock.c	2015-08-04 19:38:31.284106278 -0700
@@ -4304,7 +4304,11 @@
       goto out;
    }

+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
    memcpy_fromiovec(VMCI_DG_PAYLOAD(dg), msg->msg_iov, len);
+#else
+   memcpy_from_msg(VMCI_DG_PAYLOAD(dg), msg, len);
+#endif

    dg->dst = VMCI_MAKE_HANDLE(remoteAddr->svm_cid, remoteAddr->svm_port);
    dg->src = VMCI_MAKE_HANDLE(vsk->localAddr.svm_cid, vsk->localAddr.svm_port);
@@ -4664,8 +4668,13 @@
        * able to send.
        */

+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
       written = vmci_qpair_enquev(vsk->qpair, msg->msg_iov,
                                   len - totalWritten, 0);
+#else
+      written = vmci_qpair_enquev(vsk->qpair, &msg->msg_iter.iov,
+                                  len - totalWritten, 0);
+#endif
       if (written < 0) {
          err = -ENOMEM;
          goto outWait;
@@ -4723,6 +4732,9 @@
    VMCIDatagram *dg;
    size_t payloadLen;
    struct sk_buff *skb;
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 19, 0)
+   struct iov_iter to;
+#endif

    sk = sock->sk;
    noblock = flags & MSG_DONTWAIT;
@@ -4761,7 +4773,12 @@
    }

    /* Place the datagram payload in the user's iovec. */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
    err = skb_copy_datagram_iovec(skb, sizeof *dg, msg->msg_iov, payloadLen);
+#else
+   iov_iter_init(&to, READ, msg->msg_iter.iov, 1, payloadLen);
+   err = skb_copy_datagram_iter(skb, 0, &to, payloadLen);
+#endif
    if (err) {
       goto out;
    }
@@ -4907,9 +4924,17 @@
          }

          if (flags & MSG_PEEK) {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
             read = vmci_qpair_peekv(vsk->qpair, msg->msg_iov, len - copied, 0);
+#else
+            read = vmci_qpair_peekv(vsk->qpair, &msg->msg_iter.iov, len - copied, 0);
+#endif
          } else {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(3, 19, 0)
             read = vmci_qpair_dequev(vsk->qpair, msg->msg_iov, len - copied, 0);
+#else
+            read = vmci_qpair_dequev(vsk->qpair, &msg->msg_iter.iov, len - copied, 0);
+#endif
          }

          if (read < 0) {
